backend/danswer/db/standard_answer.py

```diff
@@ -0,0 +1,191 @@
+        db_session=db_session,
+    )
+    if len(existing_categories) != len(categories):
+        raise ValueError(f"Some or all categories with ids {categories} do not exist")
+
+    standard_answer = StandardAnswer(
+        keyword=keyword,
+        answer=answer,
+        categories=existing_categories,
+    )
+    db_session.add(standard_answer)
+    db_session.commit()
+    return standard_answer
+
+
+def update_standard_answer(
+    standard_answer_id: int,
+    keyword: str,
+    answer: str,
+    categories: list[int],
+    db_session: Session,
+) -> StandardAnswer:
+    standard_answer = db_session.scalar(
+        select(StandardAnswer).where(StandardAnswer.id == standard_answer_id)
+    )
+    if standard_answer is None:
+        raise ValueError(f"No standard answer with id {standard_answer_id}")
+
+    existing_categories = fetch_standard_answer_categories_by_ids(
+        standard_answer_category_ids=categories,
+        db_session=db_session,
+    )
+    if len(existing_categories) != len(categories):
+        raise ValueError(f"Some or all categories with ids {categories} do not exist")
+
+    standard_answer.keyword = keyword
+    standard_answer.answer = answer
+    standard_answer.categories = existing_categories
+
+    db_session.commit()
+
+    return standard_answer
+
+
+def remove_standard_answer(
+    standard_answer_id: int,
+    db_session: Session,
+) -> None:
+    standard_answer = db_session.scalar(
+        select(StandardAnswer).where(StandardAnswer.id == standard_answer_id)
+    )
+    if standard_answer is None:
+        raise ValueError(f"No standard answer with id {standard_answer_id}")
+
+    db_session.delete(standard_answer)
+    db_session.commit()
+
+
+def update_standard_answer_category(
+    standard_answer_category_id: int,
+    category_name: str,
+    db_session: Session,
+) -> StandardAnswerCategory:
+    standard_answer_category = db_session.scalar(
+        select(StandardAnswerCategory).where(
+            StandardAnswerCategory.id == standard_answer_category_id
+        )
+    )
+    if standard_answer_category is None:
+        raise ValueError(
+            f"No standard answer category with id {standard_answer_category_id}"
+        )
+
+    if not check_category_validity(category_name):
+        raise ValueError(f"Invalid category name: {category_name}")
+
+    standard_answer_category.name = category_name
+
+    db_session.commit()
+
+    return standard_answer_category
+
+
+def fetch_standard_answer_category(
+    standard_answer_category_id: int,
+    db_session: Session,
+) -> StandardAnswerCategory | None:
+    return db_session.scalar(
+        select(StandardAnswerCategory).where(
+            StandardAnswerCategory.id == standard_answer_category_id
+        )
+    )
+
+
+def fetch_standard_answer_categories_by_ids(
+    standard_answer_category_ids: list[int],
+    db_session: Session,
+) -> list[StandardAnswerCategory]:
+    return db_session.scalars(
```

Comment:
Typing issue here, may need to cast to list or return Sequence

Reply:
By the way, don't mind the other mypy issues in the automatic checks, some contributor code wasn't clean but I'm fixing it so no need to duplicate work